src/routes/GroupSettings/GroupSettingsTab/GroupSettingsTab.js

```diff
@@ -92,9 +94,15 @@ export default class GroupSettingsTab extends Component {
   updateSettingDirectly = (key, changed) => value =>
     this.updateSetting(key, changed)({ target: { value } })
 
-  save = () => {
+  save = async () => {
     this.setState({ changed: false })
-    this.props.updateGroupSettings(this.state.edits)
+    const { group, fetchLocation } = this.props
+    let coordinateLocationId
+    if (group && this.state.edits.location !== group.location) {
+      coordinateLocationId = await ensureLocationIdIfCoordinate({ fetchLocation, location: this.state.edits.location, locationId: this.state.edits.locationId })
```

Comment:
again why not do this all in the polling? otherwise we are blocking save with another server call